cos-fleet-catalog-tools/camel-connector-maven-plugin/pom.xml

```diff
@@ -498,6 +498,197 @@
                             </bannedDefinitions>
                         </configuration>
                     </execution>
+                    <execution>
+                        <id>generate-processors-yaml-schema-kebab-case-restricted</id>
+                        <phase>generate-sources</phase>
+                        <goals>
+                            <goal>generate-yaml-schema</goal>
+                        </goals>
+                        <configuration>
+                            <kebabCase>true</kebabCase>
+                            <outputFile>src/generated/resources/schema/processors-camel-yaml-dsl-restricted.json</outputFile>
+                            <additionalProperties>false</additionalProperties>
+                            <bannedDefinitions>
```

Comment:
I guess this list of banned definitions still need to be refined right ?

Reply:
@lburgazzoli For now my plan was to have a `make` task in `kas-fleet-manager` to pull this file for use in the Control Plane (see my comment to that affect in this PR). @robpblake has been working on what a catalog may be for Processors but that too is static file based. Implementation of a proper catalog for Processors is probably reasonable for follow up work. It was not covered by the Epic Brief.

Secondly, yes; the allowed definitions are yet to be agreed. I simply enabled enough to support the examples currently given for routes in the Epic Brief.

Reply:
> @lburgazzoli For now my plan was to have a `make` task in `kas-fleet-manager` to pull this file for use in the Control Plane (see my comment to that affect in this PR). @robpblake has been working on what a catalog may be for Processors but that too is static file based. Implementation of a proper catalog for Processors is probably reasonable for follow up work. It was not covered by the Epic Brief.
> 

Just to understand it correctly, the current implementation of connector catalogs is still based on static files, they are just provided as a configmap: are we following the same approach or are we thinking to embed the schema at this stage ?

Reply:
@lburgazzoli I was thinking of embedding the single processor schema we have.

 See https://github.com/bf2fc6cc711aee1a0c2a/cos-fleet-catalog-camel/pull/641#issuecomment-1489990533

Reply:
This PR is all about "initial run". 

The list of banned definitions still needs to be defined.

I have created a follow-up JIRA: https://issues.redhat.com/browse/MGDCTRS-2228

Reply:
> we really need to think about upgrades and what not.

@robpblake is working on support for a "Catalog for Processors".